src/nirizan/metrics/stats.py

```
@@ -0,0 +1,159 @@
+    Args:
+        scores: Input score sequence or array.
+
+    Returns:
+        Validated 1D float NumPy array.
+
+    Raises:
+        ValueError: If array is empty, contains NaNs/Infs, or falls outside [0, 1].
+    """
+    arr = np.asarray(scores, dtype=float)
+    if arr.ndim != 1:
+        raise ValueError("Scores must be one-dimensional.")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)